cloudify_agent/shell/commands/daemons.py

```diff
@@ -386,3 +388,39 @@ def _parse_custom_options(options):
         parsed[key] = value
 
     return parsed
+
+
+def _get_manager_ip(attributes):
+    """Calculate the correct manager IP from a list of available IPs
+    """
+    click.echo('Attempting to calculate manager IP')
+    broker_port = _get_broker_port(attributes)
+    # The manager ips var is a comma delimited string of IPs (as it passes
+    # through env variables) representing a list - so we split
+    manager_ips = attributes.pop('manager_ips')
+    manager_ips = manager_ips.split(',')
+    if not manager_ips:
+        raise exceptions.DaemonMissingMandatoryPropertyError(
+            'Passed empty list of manager IPs'
+        )
+    for host in manager_ips:
+        sock = socket.socket()
+        try:
+            # Try to connect to the host, and return it if successful
+            sock.connect((host, broker_port))
```

Comment:
in case of success, you won't get there (you have return..)
indeed connect can take a while, but it doesn't say that it's broken,
probably timeout should be configurable..